lib/javascripts/base_app.js

```diff
@@ -157,8 +157,16 @@ BaseApp.prototype = {
   },
 
   // https://developer.zendesk.com/apps/docs/agent/promises
-  when: function() {
-    return $.when.apply($, arguments);
+  when: function(...promises) {
+    return $.when(...promises.map(promise => {
+      // convert native promises to jQuery promises. See https://stackoverflow.com/a/36255846
+      if (_.isObject(promise) && !$.isFunction(promise.promise) && $.isFunction(promise.then)) {
```

Comment:
Any reason for using jQuerys `isFunction` vs lodash/underscores? Looks like the `_` method is used elsewhere